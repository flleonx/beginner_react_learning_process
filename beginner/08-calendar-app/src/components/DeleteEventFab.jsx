import { useDispatch } from 'react-redux';

import { calendarEventStartDelete } from '../actions/calendarEvent';

export const DeleteEventFab = () => {

  const dispatch = useDispatch();

  const handleDelete = () => {
    dispatch(calendarEventStartDelete());
  };

  return (
    <button
      className="btn btn-danger fab-danger"
      onClick={handleDelete}
    >
      <i className="fas fa-trash"></i>
      <span> Eliminar evento</span>
    </button>
  )
};
