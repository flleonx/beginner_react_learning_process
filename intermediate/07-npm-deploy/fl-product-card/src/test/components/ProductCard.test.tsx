import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../components';
import { product1 } from '../data/products';

const { act } = renderer;

describe('ProductCard', () => {

  test('It should show <ProductCard /> properly with the custom title', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {() => (<h1>ProductCard</h1>)}
      </ProductCard>
    )
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('It should increment the counter', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {({count, increaseBy}) => (
          <>
            <h1>ProductCard</h1>
            <span>{count}</span>
            <button onClick={() => increaseBy(1)}></button>
          </>
        )}
      </ProductCard>
    )

    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();

    act(() => {
      (tree as any).children[2].props.onClick();
    });

    tree = wrapper.toJSON();

    expect((tree as any).children[1].children[0]).toBe('1');

  });

});
