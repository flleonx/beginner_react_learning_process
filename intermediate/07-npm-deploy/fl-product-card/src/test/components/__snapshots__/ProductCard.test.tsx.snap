// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard It should increment the counter 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    ProductCard
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`ProductCard It should show <ProductCard /> properly with the custom title 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    ProductCard
  </h1>
</div>
`;
